frontend/src/components/pages/agents/details/send-message-via-a2a.tsx

```diff
@@ -0,0 +1,393 @@
+    if (evt.kind === 'message') {
+      const messageText = extractMessageText(evt);
+      if (messageText && !taskId) {
+        // For non-task messages, just concatenate
+        taskPlan += messageText;
+        if (onStreamUpdate) {
+          await onStreamUpdate(taskPlan);
+        }
+      }
+      continue;
+    }
+
+    // Handle artifact updates
+    if (evt.kind === 'artifact-update') {
+      const artifactText = extractArtifactText(evt);
+      if (artifactText) {
+        lastAgentMessage = artifactText;
+        if (onStreamUpdate && taskPlan) {
+          await onStreamUpdate(taskPlan, reasoningSteps.join('\n\n'));
+        }
+      }
+    }
+  }
+
+  // Return the result with metadata
+  const result: StreamingResult = {
+    message: taskPlan || lastAgentMessage || 'Agent completed task successfully',
+  };
+
+  if (taskId) {
+    result.taskId = taskId;
+  }
+
+  if (reasoningSteps.length > 0) {
+    result.reasoning = reasoningSteps.join('\n\n');
+  }
+
+  // If we have a final agent message that's different from the task plan and we saw final status,
+  // include it as a separate completion message
+  if (hasSeenFinalStatus && lastAgentMessage && lastAgentMessage !== taskPlan && !isToolMessage(lastAgentMessage)) {
+    result.completionMessage = lastAgentMessage;
+  }
+
+  return result;
+}
+
+/**
+ * Sends a non-streaming message and returns the response
+ */
+async function sendNonStreamingMessage(client: A2AClient, sendParams: MessageSendParams): Promise<A2AApiResponse> {
+  const response = await client.sendMessage(sendParams);
+
+  if ('error' in response) {
+    const errorMessage = response.error?.message || 'Unknown error from agent';
+    return {
+      success: false,
+      message: errorMessage,
+      error: errorMessage,
+    };
+  }
+
+  const responseText = extractTextFromResult(response.result);
+  return {
+    message: responseText,
+    success: true,
+  };
+}
+
+const fetchWithCustomHeader: typeof fetch = async (url, init) => {
+  const headers = new Headers(init?.headers);
+  headers.set('Authorization', `Bearer ${config.jwt}`);
+
+  const newInit = { ...init, headers };
+
+  return fetch(url, newInit);
+};
+
+/**
+ * Sends a message to an AI agent via the A2A (Agent-to-Agent) protocol.
+ * Uses the official @a2a-js/sdk for proper A2A protocol communication.
+ * Supports streaming responses from agents.
+ *
+ * @see https://github.com/a2aproject/a2a-js for SDK documentation
+ */
+export const sendMessageViaA2A = async ({
+  message,
+  agentUrl,
+  onStreamUpdate,
+}: SendMessageViaA2AProps): Promise<A2AApiResponse> => {
+  try {
+    // Determine the agent card URL
+    // If agentUrl already includes /.well-known/agent.json, use it as-is
+    // Otherwise, append the standard path
+    const agentCardUrl =
+      agentUrl.includes('agent.json') || agentUrl.includes('well-known')
+        ? agentUrl
+        : `${agentUrl}/.well-known/agent.json`;
```

Comment:
blocker: what are the options here ? It's not very obvious which urls will be valid or invalid, why would I have a different URL. and which URL are consider valid ?